### DISCUSSION TITLE
Does Remix proprietary express serve serve using HTTP/1 by default?

### DISCUSSION THREAD

**Discussion by joeprivettsmart** [remix-run/remix#9130] [Q&A] [answered]
We are seeing our JS bundles being served using HTTP/1, so just wanted to check if Remix express server is doing this by default and, is so, why not HTTP/2?

**Comment by alexanderson1993** [comment]
Remix isn't an HTTP server, it's just a request handler. So whether the request is served from an HTTP/1 or HTTP/2 server is up to Express, or whatever other HTTP server you're using.

I can only hazard a guess as to why - HTTP/1 is more broadly supported and is easier to set up overall. For example, HTTP/2 requires TLS certificates, which can make local dev annoying. To make onboarding easier, Remix defaults to HTTP/1, which works for most use-cases.

**Reply by joeprivettsmart** [reply]
This seems to suggest that Remix server is using express:
https://remix.run/docs/en/main/other-api/serve

**Reply by alexanderson1993** [reply, marked as answer]
Yes, the built-in server that Remix's CLI provides for convenience is using express. But, they specifically say that it's basic and you should use their Express adapter (or another adapter) if you want to do anything more sophisticated. I would say HTTP/2 falls into that category.

> Remix is designed for you to own your server, but if you don't want to set one up you can use the Remix App Server instead. It's a production-ready, but basic Node.js server built with Express. If you find you want to customize it, use the @remix-run/express adapter instead.
